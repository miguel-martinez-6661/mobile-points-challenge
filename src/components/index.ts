export * from './text/text';
export * from './button/button';
export * from './card/card';
export * from './loading-indicator/loading-indicator';
