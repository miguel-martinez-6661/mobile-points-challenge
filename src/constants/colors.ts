// These colors should be defined in a design system and included in the theme file.
// This is just a quick example.
export enum Colors {
  primary = '#334FFA',
  secondary = '#CFD6FF',
  tertiary = '#9B9898',
  success = '#00B833',
  error = '#FF0000',
  background = '#FFFFFF',
  defaultBackground = '#f2f2f2',
  textPrimary = '#000000',
  textSecondary = '#FFFFFF',
  textTertiary = '#9B9898',
}
