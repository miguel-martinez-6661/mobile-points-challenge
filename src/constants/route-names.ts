export enum RouteNames {
  Home = 'Home',
  Details = 'Details',
}
