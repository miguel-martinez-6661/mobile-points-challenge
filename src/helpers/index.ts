export * from './numbers-helpers';
export * from './date-helpers';
