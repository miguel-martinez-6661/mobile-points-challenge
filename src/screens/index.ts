export * from './home/home-screen';
export * from './details/product-details-screen';
