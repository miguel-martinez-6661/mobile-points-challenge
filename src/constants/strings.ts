// This file contains all the strings used in the app.
// In case translations are needed, this is the where all keys are defined.
import { RouteNames } from './route-names';

export const strings = {
  [RouteNames.Home]: {
    greetings: 'Bienvenido de vuelta',
    yourPoints: 'TUS PUNTOS',
    yourPointsFlow: 'TUS MOVIMIENTOS',
    allButton: 'Todos',
    earnedPointsButton: 'Ganados',
    redeemedPointsButton: 'Canjeados',
  },
  [RouteNames.Details]: {
    productDetails: 'Detalles del producto',
    purchasePointsEarned: 'Con esta compra acumulaste',
    purchasedOn: 'Comprado el',
    points: 'puntos',
    acceptButton: 'Aceptar',
  },
};
