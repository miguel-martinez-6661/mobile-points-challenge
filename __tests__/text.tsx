import 'react-native';
import React from 'react';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import { Text } from '../src/components';
import { Colors } from '../src/constants/colors';

describe('Testing <Text> component', () => {
  it('should mount component correctly', () => {
    const textComponent = renderer.create(<Text>Test</Text>).toJSON();
    expect(textComponent).toMatchSnapshot();
  });

  it('should render with "title" props', () => {
    const textComponent = renderer.create(<Text title>Test</Text>).toJSON();
    expect(textComponent).toMatchSnapshot();
  });

  it('should render with "subTitle" props', () => {
    const textComponent = renderer.create(<Text subTitle>Test</Text>).toJSON();
    expect(textComponent).toMatchSnapshot();
  });

  it('should render with "largeText" props', () => {
    const textComponent = renderer.create(<Text largeText>Test</Text>).toJSON();
    expect(textComponent).toMatchSnapshot();
  });

  it('should support color changing by passing a color code as prop', () => {
    const textPrimaryComponent = renderer
      .create(<Text color={Colors.textPrimary}>Test</Text>)
      .toJSON();

    const textSecondaryComponent = renderer
      .create(<Text color={Colors.textSecondary}>Test</Text>)
      .toJSON();

    const textTertiaryComponent = renderer
      .create(<Text color={Colors.textTertiary}>Test</Text>)
      .toJSON();

    expect(textPrimaryComponent).toMatchSnapshot();
    expect(textSecondaryComponent).toMatchSnapshot();
    expect(textTertiaryComponent).toMatchSnapshot();
  });

  it('should support custom styles by receiving a style objects as a prop', () => {
    const textComponent = renderer
      .create(<Text styles={{ fontSize: 14 }}>Test</Text>)
      .toJSON();
    expect(textComponent).toMatchSnapshot();
  });
});
