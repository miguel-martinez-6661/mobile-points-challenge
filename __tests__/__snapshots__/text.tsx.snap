// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <Text> component should mount component correctly 1`] = `
<Text
  style={
    Object {
      "color": "#000000",
      "fontFamily": "Avenir",
      "fontSize": 18.857142857142858,
      "fontWeight": "400",
    }
  }
>
  Test
</Text>
`;

exports[`Testing <Text> component should render with "largeText" props 1`] = `
<Text
  style={
    Object {
      "color": "#000000",
      "fontFamily": "Avenir",
      "fontSize": 37.714285714285715,
      "fontWeight": "400",
    }
  }
>
  Test
</Text>
`;

exports[`Testing <Text> component should render with "subTitle" props 1`] = `
<Text
  style={
    Object {
      "color": "#000000",
      "fontFamily": "Avenir",
      "fontSize": 22,
      "fontWeight": "400",
    }
  }
>
  Test
</Text>
`;

exports[`Testing <Text> component should render with "title" props 1`] = `
<Text
  style={
    Object {
      "color": "#000000",
      "fontFamily": "Avenir",
      "fontSize": 28.285714285714285,
      "fontWeight": "400",
    }
  }
>
  Test
</Text>
`;

exports[`Testing <Text> component should support color changing by passing a color code as prop 1`] = `
<Text
  style={
    Object {
      "color": "#000000",
      "fontFamily": "Avenir",
      "fontSize": 18.857142857142858,
      "fontWeight": "400",
    }
  }
>
  Test
</Text>
`;

exports[`Testing <Text> component should support color changing by passing a color code as prop 2`] = `
<Text
  style={
    Object {
      "color": "#FFFFFF",
      "fontFamily": "Avenir",
      "fontSize": 18.857142857142858,
      "fontWeight": "400",
    }
  }
>
  Test
</Text>
`;

exports[`Testing <Text> component should support color changing by passing a color code as prop 3`] = `
<Text
  style={
    Object {
      "color": "#9B9898",
      "fontFamily": "Avenir",
      "fontSize": 18.857142857142858,
      "fontWeight": "400",
    }
  }
>
  Test
</Text>
`;

exports[`Testing <Text> component should support custom styles by receiving a style objects as a prop 1`] = `
<Text
  style={
    Object {
      "color": "#000000",
      "fontFamily": "Avenir",
      "fontSize": 14,
      "fontWeight": "400",
    }
  }
>
  Test
</Text>
`;
